pyproject.toml

```diff
@@ -0,0 +1,157 @@
+  "imap_processing/tests",
+]
+addopts = "-ra"
+filterwarnings = [
+    "ignore:Converting non-nanosecond:UserWarning:cdflib",
+    "ignore:datetime.datetime.utcfromtimestamp:DeprecationWarning:cdflib",
+    # Ignore Data_version warnings during tests because we default to v001
+    "ignore:No Data_version attribute found:UserWarning:imap_processing",
+]
+markers = [
+    "external_kernel: marks tests as requiring external SPICE kernels (deselect with '-m \"not external_kernel\"')",
+    "external_test_data: marks tests as requiring external test data (deselect with '-m \"not external_test_data\"')",
+    "use_test_metakernel: Mark test to use a test metakernel"
+]
+
+
+[tool.ruff]
+target-version = "py39"
+lint.select = ["B", "D", "E", "F", "I", "N", "S", "W", "PL", "PT", "UP", "RUF"]
+lint.ignore = [
+  "D104",  # Missing docstring in public package
+  "PLR0913",  # too-many-args error
+  "PLR0915",  # too-many-statements error
+  "PLR2004",  # Magic value in comparison
+  "RUF002",  # `−` (MINUS SIGN). Did you mean `-` (HYPHEN-MINUS) error
+  "RUF200",  # pyproject missing field (poetry doesn't follow the spec)
+  "S311",  # suspicious-non-cryptographic-random-usage
+]
+
+[tool.ruff.lint.per-file-ignores]
+# S603 unchecked input in subprocess call is fine in our tests
+# PT006 Wrong type passed to first argument of @pytest.mark.parametrize
+"*/tests/*" = ["D", "S101", "S603", "PT006"]
+
+[tool.ruff.lint.pydocstyle]
+convention = "numpy"
+
+[tool.poetry.scripts]
+imap_cli = 'imap_processing.cli:main'
+imap_xtce = 'imap_processing.ccsds.excel_to_xtce:main'
+
+[tool.codespell]
+ignore-words-list = "livetime"
+
+[tool.poetry-dynamic-versioning]
+enable = true
+vcs = "git"
+
+[tool.poetry-dynamic-versioning.files."imap_processing/_version.py"]
+persistent-substitution = true
+initial-content = """
+  # These version placeholders will be replaced later during substitution.
+  __version__ = "0.0.0"
+  __version_tuple__ = (0, 0, 0)
+"""
+
+[tool.numpydoc_validation]
+checks = ["all", #report on all checks, except the following
+    "EX01", # Ignore No examples section found
+    "SA01", # Ignore See Also section not found
+    "ES01", # Ignore No extended summary found
+    "RT02" ] # Ignore The first line of the Returns section
+exclude = ['__init__' ] # don't report on objects that match any of these regex
+override_SS05 = [  # override SS05 to allow docstrings starting with these words
+    '^Process ',
+]
+
+[tool.mypy]
+disable_error_code = ['import-not-found', # Unable to locate module or package specified
+                'import-untyped', # The import lacks type information
+                'no-untyped-call', # Function calls are only made to functions that are fully typed
+                'type-arg', # Requires type arguments to be specified in list[] and dict[]
+                'union-attr', # Item "float" of "float | Any" has no attribute "reshape"
+                'no-any-return', # Returning Any from a function
+              ]
```

Comment:
Similar to my comment in the `ruff` section above, feel free to change these as best fit for this repo.